docs/development/i18n-guide.md

```diff
@@ -0,0 +1,261 @@
+# Renovate i18n Guide
+
+This document briefs how to provide translatable to user-facing text, how to contribute code for i18n, and what is your recommended way/style to do that.
```

Comment:
Let's split our i18n documentation into two parts:

- Things developers of Renovate (who may not care much about translation) need to know
- What those who care about translating need to know

Reply:
Since we already put the translating relevant code to a [separated repository](https://github.com/xingxing/renovate-i18n).
I think whether we can only keep the  "Things developers of Renovate (who may not care much about translation) need to know" document here, and  put the translators's  guide  the renovate-i18n repository?

Reply:
Right now this document is 155 lines long. If this is the "minimum things a developer needs to know about i18n to contribute a Renovate PR" then it's too much. We really need a "keep it simple" guide for the 99% of developers who won't translate anything, and just need to know how to make sure anything they contribute is compatible